File: MyBackEnd/src/main/java/com/example/mybackend/Models/Employee.java
package com.example.mybackend.Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private double salary;
    private String address;
    private String phone;
    private String email;
    private String gender;
    private String image ;
    private String shift;
    private LocalDate HiringDate;
    private Boolean isActive;

    @ManyToOne
    private position position; // Link to Position entity

    @ManyToOne
    private Department department; // Link to Department entity

}

File: WebTs/src/types/Employee.ts
import { Department } from "./Departement";
import { Position } from "./position";

export interface Employee {
  id: number;
  name: string;
  position: Position;
  salary: number;
  department: Department;
  address: string;
  phone: string;
  email: string;
  gender: string;
  image: string;
  shift: string;
  hiringDate: Date;
}
export const emptyEmployee: Employee = {
  id: 0,
  name: "",
  position: {
    id: 0,
    name: "",
    level: "",
    responsibilities: "",
    qualifications: "",
  },
  salary: 0,
  department: {
    id: 0,
    name: "",
    description: "",
  },
  address: "",
  phone: "",
  email: "",
  gender: "",
  image: "",
  shift: "",
  hiringDate: new Date(),
};
